tests/CMakeLists.txt

```diff
@@ -0,0 +1,10 @@
+enable_testing()
+
+file(GLOB_RECURSE all_tests ${PROJECT_SOURCE_DIR}/tests/*.c)
+
+foreach(filename ${all_tests})
+  get_filename_component(testname ${filename} NAME_WE)
+  add_executable(${testname} ${filename})
+  target_link_libraries(${testname} PUBLIC CEED)
+  add_test(${testname} ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/${testname})		
```

Comment:
Does this run for you? The test executables take the backend as an argument. (Mine fails to link.)

Reply:
I wasn't able to get the tests to run initially (see https://github.com/CEED/libCEED/pull/1525#issuecomment-2010346893), but they did compile successfully on my mac and ubuntu workstation.

@jeremylt 's comment about which arguments to pass to the tests was helpful, although for some reason CMake's built in test fixtures pass arguments in quotations, e.g.

```
test000 "/ref/opt/cuda"
```

as opposed to

```
test000 /ref/opt/cuda
```
which seems to mess up how the tests parse which backends to use.

I'm less familiar with that corner of CMake, so some potential workarounds are to slightly generalize how the tests read the input arguments, or to call the tests through a bash script that strips the extra quotes.